src/commands/TranformCamelRouteCommand.ts

```diff
@@ -0,0 +1,105 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License", destination); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict'
+
+import { commands, Uri, window, workspace, WorkspaceFolder } from 'vscode';
+import { CamelTransformRouteToYAMLJBangTask } from '../tasks/CamelTransformRouteToYAMLJBangTask';
+import path = require('path');
+import validFilename = require('valid-filename');
+
+export class TransformCamelRouteToYAMLCommand {
```

Comment:
if you will use `extends AbstractNewCamelRouteCommand `, you will be able to reuse methods from extended class such as 

- getWorkspaceFolder
- validateCamelFileName
- computeFullPath

Reply:
I'll do that in another PR so that I don't change the logic too much in this one.